Clarify Private Link steps (#34)
redpanda-data/cloud-docs: modules/networking/pages/azure-private-link.adoc

Replace these placeholder variables:

* `<redpanda-cluster-subscription-id>`: The ID of the subscription where the Redpanda cluster will be provisioned.
* `<source-connection-subscription-id>`: The ID of the subscription from where you will initiate connections to the Private Link service.

> In most cases, you should use two distinct subscription IDs.

Author: @c4milo is there any reason or case for using the same subscription here?

Reviewer: Any subscription can be used.

Reviewer: No need to say "In most cases..."

Replace these placeholder variables:

> * `<redpanda-cluster-subscription-id>`: The ID of the subscription where the Redpanda cluster will be provisioned.

Reviewer 2: ```suggestion
* `<redpanda-cluster-subscription-id>`: The ID of the subscription where the Redpanda cluster is provisioned.
```

> * `<source-connection-subscription-id>`: The ID of the subscription from where you will initiate connections to the Private Link service.

Reviewer 2: ```suggestion
* `<source-connection-subscription-id>`: The ID of the subscription from where you initiate connections to the Private Link service.
```

=== Set up Azure Private Link endpoint in your virtual network

> . If you have not already done so, create the Azure resource group and virtual network for your Private Link source connections. This virtual network for the Private Link endpoint cannot be the same as the virtual network for the Redpanda cluster.

Reviewer: Sorry for the back and forth. I think it is better not to say this as technically speaking, it is valid for them to be in the same VNet as long as the private link endpoint is not on the same subnet as the private link service. So, to avoid creating confusion, I think it is better to remove this bit.

    --registration-enabled false
```

> . Create a DNS zone group, then create a wildcard record in the private DNS zone.

Author: @c4milo @ivpanda Should this be broken into two separate steps? In Ivana's testing it sounds like a wildcard record is required either way. Can we say that the DNS zone group is optional?

Reviewer: Let's remove the creation of that DNS group for now.

Reviewer: I thought it was going to spare us from creating the DNS record but in Ivana's test it didn't.
